javascript/q2.js

```diff
@@ -0,0 +1,34 @@
+function Metadataparser(version, channel, keyField) {
+    this._version = version
+    this._channel = channel
+    this._keyField = keyField
+
+    this.getVersion = function() {
```

Comment:
use the getters and setters.
You may refer:
https://www.programiz.com/javascript/getter-setter
https://javascript.info/property-accessors

Reply:
Not applicable for constructor function